### DISCUSSION TITLE
Question on declaring @anyOf on Named Unions

### DISCUSSION THREAD

**Discussion by Devon-White** [microsoft/typespec#4387] [OpenAPI3] [answered]
I'm finding that declaring a Named Union with the `@oneOf` decorator doesn't seem to enforce `anyOf` becoming `oneOf` is the spec file when used as a value in a Model.

I also tried decorating the property in the Model that accepts the Named Union as the value with `oneOf`, same behavior.

Example can be found [here](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7DQrSGm9wZW5hcGkzxB4NCnVzaW5nIFR5cGVTcGVjLk9wZW5BUEnFGw0KbW9kZWwgQSB7DQogIHRlc3Q6ICLEBzEiLMgSMscTMiINCn3KNELKNDPHIjPKNTTHEzTkAITFNkBvbmVPZg0KdW5pb24gQ2FsbFVwZGF0ZVBhcmFtc0FsaWFzIHsgQSB8IEJ95QCdxmfKKVJlcXVlc3TGdyAgY29tbWFuZDogInXFH8RpxBjIaMQMcMVcOtZu6ACZDQo%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)

However, removing the Name Union and using a Union Expression inside the model, will allow it to generate with `oneOf`. 

Example can be found [here](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7DQrSGm9wZW5hcGkzxB4NCnVzaW5nIFR5cGVTcGVjLk9wZW5BUEnFGw0KbW9kZWwgQSB7DQogIHRlc3Q6ICLEBzEiLMgSMscTMiINCn3KNELKNDPHIjPKNTTHEzTkAITFNso6Q2FsbFVwZGF0ZVJlcXVlc3TGSiAgY29tbWFuZDogInXFH8Q8xBhAb25lT2bGDHBhcmFtczogQSB8IELKXA%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)

Should the first example work?

**Comment by timotheeguerin** [comment, marked as answer]
I think in the first case it might be a confusion in the syntax that result in not exactly what you expect 

In named union the variants are separated with comma `,`. What you have is a union of model `A` and `B` inside that named union. So when openapi3 sees it, it sees a union with a single variant `A | B` so just ignore anyOf or oneOf and just render the single variant which happens to be a union which is then not annotated with `@oneOf`


[See this playground](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7DQrSGm9wZW5hcGkzxB4NCnVzaW5nIFR5cGVTcGVjLk9wZW5BUEnFG0BvbmVPZg0KdW5pb24gT8QNVcULew0KICBzdHJpbmcsxAtpbnQzMiwNCn3SNsUxSW7ROCB8zDY%3D&e=%40typespec%2Fopenapi3&options=%7B%7D) showing the difference
